.github/setup_python/action.yml

```diff
@@ -0,0 +1,27 @@
+name: "Setup Python"
+description: "Sets up Python for use in actions with caching and copy localsettings"
+
+runs:
+  using: "composite"
+  steps:
+    - name: Setup python
+      uses: actions/setup-python@v4
+      with:
+        python-version: 3.8
+
+    - name: Restore venv cache
+      uses: syphar/restore-virtualenv@v1
```

Comment:
A thought: Should these versions be pinned? I suppose there isn't anything bad that could happen if the author turns malicious here, so its probably fine to leave it at `v1`, but what do the others think?

Reply:
I'm fine with them not being pinned